src/platform/user/profile/vet360/containers/AddressValidationModal.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Modal from '@department-of-veterans-affairs/formation-react/Modal';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+import { selectCurrentlyOpenEditModal } from '../selectors';
+import { openModal, createTransaction } from '../actions';
+
+import * as VET360 from '../constants';
+
+class AddressValidationModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedOption: '',
+      address: {},
+    };
+  }
+  onChangeHandler = address => event => {
+    this.setState({ selectedOption: event.target.name, address });
+  };
+
+  onSubmit = () => {
+    const { validationKey } = this.props;
+    const { address } = this.state;
+    const payload = { ...address, validationKey };
+
+    const method = payload.id ? 'PUT' : 'POST';
```

Comment:
Just want to make sure this is correct. The payload in this case is whatever address the user picked from the validation modal, correct? If so, how is the payload.id being set? We want to change the submit method based on if the user already has an address on file and want to confirm that's being determined correctly.